apollo-router/src/services/layers/persisted_queries/manifest_poller.rs

```diff
@@ -207,7 +207,69 @@ impl PersistedQueryManifestPoller {
     /// Starts polling immediately and this function only returns after all chunks have been fetched
     /// and the [`PersistedQueryManifest`] has been fully populated.
     pub(crate) async fn new(config: Configuration) -> Result<Self, BoxError> {
-        if let Some(uplink_config) = config.uplink.as_ref() {
+        if let Some(local_pq_list) = config.persisted_queries.safelist.local_safelist {
+            tracing::info!(
+                "Loading persisted query list from local file: {}",
+                local_pq_list.clone()
+            );
+            let local_manifest =
+                std::fs::read_to_string(local_pq_list.clone()).map_err(|e| -> BoxError {
+                    format!("could not read local persisted query list file: {}", e).into()
+                })?;
+
+            let mut manifest = PersistedQueryManifest::new();
+
+            let manifest_file: SignedUrlChunk =
```

Comment:
We should probably check the format and version fields?

Reply:
🤦 i meant to do that. maybe worth abstracting it out into a validation function.